main/main.cpp

```diff
@@ -478,6 +478,12 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 
 	I = args.front();
 	while (I) {
+#ifdef OSX_ENABLED
+		if (I->get().begins_with("-psn_")) {
```

Comment:
```suggestion
		// Ignore the process serial number argument passed by macOS Gatekeeper.
		// Otherwise, Godot would try to open a nonexistent project on the first start and abort.
		if (I->get().begins_with("-psn_")) {
```